### ISSUE TITLE
[1.12.2] Issues with adding items to JEI

### ISSUE THREAD

**Issue by aaronhowser1** [mezz/JustEnoughItems#2053]
I'm using this line in CraftTweaker: `mods.jei.JEI.addItem(<geneticsreborn:plasmid>.withTag({gene: "GeneticsRebornSCARE_SKELETONS", num: 24, numNeeded: 24}));`
the CT log says it's working:
![](https://cdn.discordapp.com/attachments/354745056743260172/746068809378562178/unknown.png)
but it doesn't show in JEI. When I go into the console log, with JEI's debug enabled, it shows this:
![](https://cdn.discordapp.com/attachments/354745056743260172/746074921108766781/unknown.png)
Debug is supposed to show the item, right?
```java
if (Config.isDebugModeEnabled()) {
                    Log.get().debug("Added ingredient: {}", ingredientHelper.getErrorInfo(element.getIngredient()));
                }
```
![image](https://user-images.githubusercontent.com/15698722/90811930-15d9ab00-e2eb-11ea-8fc4-91675194056e.png)

**Comment by jaredlll08** [comment]
This has been an ongoing issue, I thought people would be making issues on JEI's side, but apparently not.

**Comment by mezz** [comment]
Thanks for the report!
The debug log lines will only show up in `debug.log`.
I tested this with JEI's debug ingredients and it seems to work normally.

Maybe JEI believes all `geneticsReborn:plasmid` are the same. Normally mods need to register a "subtype interpreter" with JEI to let it know how subtypes of an item can be differentiated.

Check `debug.log` and you should see `[jei]: Added ingredient:` if it's working correctly.
If you see `[jei]: Updated ingredient:` instead, that means JEI thinks all these items are the same and doesn't know how to tell them apart.

**Comment by democat3457** [comment]
Similar issue reported here (wasn't sure if I was supposed to make an issue in this repo or not so I reported it to CrT first): https://github.com/CraftTweaker/CraftTweaker/issues/1020

**Comment by Valafi** [comment]
@mezz I don't have a debug.log using Technic Launcher or Twitch. Is there something special you have to do to enable it? I already enabled debug mode in the JEI config.

**Comment by democat3457** [comment]
So this won't be fixed in 1.12? (At least as far as I've heard from Jared)

**Comment by jaredlll08** [comment]
@mezz can add the stale tag back / close it

At the end of the day, this was reported to me many times, and I told people to make an issue with JEI. 

No one did. 

Now that someone finally made the issue, when 1.12 is old news, neither Mezz nor myself want to invest time into a proper fix unfortunately.

The one potential fix we talked about, could lead to a bunch of other issues, so definitely not something to implement on an old version.
